### ISSUE TITLE
Allow space to confirm an option

### ISSUE THREAD

**Issue by tvararu** [alphagov/accessible-autocomplete#98]
While I don't believe we've seen anyone attempt this in research, the space bar is commonly used by keyboard users to confirm things. There shouldn't be a downside to adding support for this.

**Comment by edwardhorsford** [comment]
There might be a downside...

Ideally the input field would stay focused and the user can continue to type in to it - they may expect the space key to let them type a space character.

Would be good to investigate how other autocompletes handle this.

**Comment by tvararu** [comment]
@edwardhorsford Leonie's typeahead does allow you to confirm using space: http://ljwatson.github.io/design-patterns/autocomplete/index.html

So does Nicolas Hoffman's: https://a11y.nicolas-hoffmann.net/autocomplet-list/

So it seems like other autocompletes that focus on accessibility do have this behaviour.

**Comment by adamsilver** [comment]
Yes, so when (pseudo) focus is on the text box then spacebar should be a space, otherwise it should select the in focus option. This is a ARIA requirement.
